gluon-activations/gluon-activations.md

```diff
@@ -0,0 +1,239 @@
+    
+    plt.figure()
+    plt.plot(x.asnumpy(), y.asnumpy())
+    plt.plot(x.asnumpy(), x.grad.asnumpy())
+    plt.legend(["{} activation".format(activation), "{} gradient".format(activation)])
+```
+
+## Sigmoids
+
+### Sigmoid
+The sigmoid activation function, also known as the logistic function or logit function, is perhaps the most widely known activation owing to its [long history](https://web.stanford.edu/class/psych209a/ReadingsByDate/02_06/PDPVolIChapter8.pdf) in neural network training and appearance in logistic regression and kernel methods for classification. 
+
+The sigmoid activation is a non-linear function that transforms any real valued input to a value between 0 and 1, giving it a natural probabilistic interpration. The sigmoid takes the form of the function below
+
+$$ \sigma(x) = \dfrac{1}{1 + e^x} $$ or alternatively
+
+$$ \sigma(x) = \dfrac{e^x}{e^x + 1} $$
+
+Note: the term sigmoid is overloaded and can be used to refer to the class of 's' shaped functions or particularly to the logistic function that we've just described. In MxNet the sigmoid activation specifically refers to logistic function sigmoid.
+
+
+
+```python
+visualize_activation(data, gluon.nn.Activation('sigmoid'), "Sigmoid")
+```
+
+
+![png](output_5_0.png)
+
+
+The sigmoid activation has since fallen out of use as the preferred activation function in designing neural networks due to some of it's properties, shown in the plot above, like not being zero-centered and inducing vanishing gradients, that leads to poor performance during neural network training. Vanishing gradients here refers to the tendency of the gradient of the sigmoid function to be nearly zero for most input values. 
+
+### tanh
+The tanh activation function is also an s shaped curve defined by the mathematical equation:
+
+$$ tanh(x) = \dfrac{e^x - e^{-x}}{e^x + e^{-x}}$$ 
+
+tanh addresses the issues of not being zero centered associated with the sigmoid activation function but still retains the vanishing gradient problems due to the gradient being asymptotically zero for values outside a narrow range of inputs. 
+
+In fact, the tanh can be rewritten as,
+
+$$tanh(x) = \dfrac{e^{2x} - 1}{e^{2x} + 1}$$
+
+which shows it's direct relation to sigmoid by the following equation:
+
+
+
+$$ tanh(x) = 2\sigma(2x) - 1$$
+
+
+```python
+visualize_activation(data, gluon.nn.Activation('tanh'), "Tanh")
+```
+
+
+![png](output_8_0.png)
+
+
+The use tanh as activation functions in place of the logistic function was popularized by the success of the [LeNet architecture](http://yann.lecun.com/exdb/publis/pdf/lecun-01a.pdf) and the [methods paper](http://yann.lecun.com/exdb/publis/pdf/lecun-98b.pdf) by LeCun et al
+
+### SoftSign
+
+The softsign activation is an alternative to tanh that is also centered at zero but converges asymptotically to -1 and 1 polynomially instead of exponentially. As such, there are a greater range of input values for which the softsign assigns an output of between -1 and 1.
```

Comment:
As such, there are a greater range of input values for which the softsign assigns an output of between -1 and 1.
A bit confused? It's the same isn't it?

Reply:
yeah i rephrased this afterwards. what i mean is that softsign doesn't saturate as quickly as tanh. i'll find another way to say this in lay terms